sources/packages/web/src/services/http/EducationProgramApi.ts

```diff
@@ -1,118 +1,100 @@
+import { OptionItemDto, PaginationOptions } from "@/types";
+import HttpBaseClient from "./common/HttpBaseClient";
+import { getPaginationQueryString } from "@/helpers";
 import {
-  SummaryEducationProgramDto,
-  OptionItemDto,
-  EducationProgramData,
-  DataTableSortOrder,
-  ProgramSummaryFields,
-  DEFAULT_PAGE_LIMIT,
-  DEFAULT_PAGE_NUMBER,
-  PaginatedResults,
-  ApproveProgram,
-  DeclineProgram,
+  ApproveProgramAPIInDTO,
+  DeclineProgramAPIInDTO,
+  EducationProgramAPIInDTO,
+  EducationProgramAPIOutDTO,
+  EducationProgramsSummaryAPIOutDTO,
+  OptionItemAPIOutDTO,
+  PaginatedResultsAPIOutDTO,
   StudentEducationProgramAPIOutDTO,
-} from "@/types";
-import HttpBaseClient from "./common/HttpBaseClient";
-import { addSortOptions } from "@/helpers";
+} from "@/services/http/dto";
 
 export class EducationProgramApi extends HttpBaseClient {
-  public async getProgram(programId: number): Promise<any> {
-    try {
-      const response = await this.apiClient.get(
-        `institution/education-program/${programId}`,
-        this.addAuthHeader(),
-      );
-      return response.data;
-    } catch (error) {
-      this.handleRequestError(error);
-      throw error;
-    }
+  /**
+   * Get programs for a particular location with pagination.
+   * @param locationId id of the location.
+   * @param paginationOptions pagination options.
+   * @returns paginated programs summary.
+   */
+  async getProgramsSummaryByLocationId(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResultsAPIOutDTO<EducationProgramsSummaryAPIOutDTO>> {
+    const url =
+      `education-program/location/${locationId}/summary?` +
+      getPaginationQueryString(paginationOptions);
+    return this.getCallTyped<
+      PaginatedResultsAPIOutDTO<EducationProgramsSummaryAPIOutDTO>
+    >(this.addClientRoot(url));
   }
 
-  public async createProgram(createProgramDto: any): Promise<void> {
-    try {
-      await this.apiClient.post(
-        "institution/education-program",
-        createProgramDto,
-        this.addAuthHeader(),
-      );
-    } catch (error) {
-      this.handleRequestError(error);
-      throw error;
-    }
+  /**
+   * Get the programs summary of an institution with pagination.
+   * @param institutionId id of the institution.
+   * @param paginationOptions pagination options.
+   * @returns paginated programs summary.
+   */
+  async getProgramsSummaryByInstitutionId(
+    institutionId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResultsAPIOutDTO<EducationProgramsSummaryAPIOutDTO>> {
+    console.log(paginationOptions);
+    const url =
+      `education-program/institution/${institutionId}/summary?` +
```

Comment:
we can also do, 
![image](https://user-images.githubusercontent.com/77353155/181312577-680e259f-1309-4af0-b9ed-307f87efe701.png)